aws/credentials/processcreds/provider_test.go

```diff
@@ -189,6 +189,38 @@ func TestProcessProviderFromSessionWithProfileCrd(t *testing.T) {
 
 }
 
+func TestProcessProviderFromSessionWithLongSessionToken(t *testing.T) {
+	oldEnv := preserveImportantStashEnv()
+	defer awstesting.PopEnv(oldEnv)
+
+	os.Setenv("AWS_PROFILE", "long_session_token")
```

Comment:
Instead of using a profile and shared config file, this can be tested more directly:

```go
creds := processcreds.NewCredentials(
		fmt.Sprintf(
			"%s %s",
			getOSCat(),
			strings.Join(
				[]string{"testdata", "longsessiontoken.json"},
				string(os.PathSeparator))))
	_, err := creds.Get()
	if err != nil {
		t.Errorf("expected %v, got %v", "no error", err)
	}
```
This way you can get rid of the two changes to the shared config files.